outfancy/render.py

```diff
@@ -1869,6 +1882,26 @@ def insert_point(self, point, x_coord, y_coord):
         """Each element of the matrix is inserted in the window."""
         self.content[y_coord][x_coord] = point
 
+    def insert_colored_point(self, point, x_coord, y_coord):
+
+        def fidx(x,y):
+            return y * self.h + x
+
+        prev_color = ''
+
+        flat_content = [j for i in self.content for j in i]
+
+        index = fidx(x_coord,y_coord)
+
```

Comment:
This is the important function, it adds the new point to the window content, but it figures what is the last ANSI escape code used and appends it at the beginning and end of the new point.